sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/UnivocityParser.scala

```diff
@@ -110,7 +110,11 @@ private[csv] class UnivocityParser(
       }
 
     case _: BooleanType => (d: String) =>
-      nullSafeDatum(d, name, nullable, options)(_.toBoolean)
+      nullSafeDatum(d, name, nullable, options) { datum =>
+        if (datum.equalsIgnoreCase(options.trueValue)) true
```

Comment:
Is there any reason why you compare `datum` to `options.trueValue` and not vice versa as it's in `CSVInferSchema.scala` above?